docs/explanation/integration.md

```diff
@@ -0,0 +1,134 @@
+
+The plan resource output will be named the following:
+
+`<generated-id>__build__output__charm__<detected-charm-name>`
+
+### Rock
+
+The workflow will automatically look for any `rockcraft.yaml` file within the
+source repository root by using the following glob syntax: `**/rockcraft.yaml`.
+
+The plan resource output will be named the following:
+
+`<generated-id>__build__output__rock__<detected-rock-name>`
+
+### Docker
+
+The workflow will automatically look for any `.Dockerfile` file within the
+source repository root by using the following glob syntax: `**/*.Dockerfile`.
+
+The plan resource output will be named the following:
+
+`<generated-id>__build__output__docker-image__<*.Dockerfile prefix>`
+
+### Charm Resources
+
+If a charm has [file resources](https://canonical-charmcraft.readthedocs-hosted.com/en/stable/reference/files/charmcraft-yaml-file/#resources) the workflow will assume 
+`build-<resourceName>.sh` script to exist which is used to build the resource.
+
+The plan resource output will be named the following:
+
+`<generated-id>__build__output__<charm-name>__<resource-name>`
+
+## Build the artifacts
+
+### Charm
+
+The charms detected from the [plan step](#1-plan) is built using the
+`charmcraft pack` command. The workflow automatically changes the working
+directory to where the `charmcraft.yaml` or the `metadata.yaml` file resides.
+The built charm output are selected using the `*.charm` glob syntax. The output
+artifact is then uploaded to the GitHub.
+
+### Rock
+
+The rocks detected from the [plan step](#1-plan) is built using the
+`rockcraft pack` command. The workflow automatically changes the working
+directory to where the `rockcraft.yaml` file resides.
+The built rock output are selected using the `*.rock` glob syntax. The output
+artifact is then uploaded to the GitHub or uploaded to the Docker repository.
+
+### Docker
+
+The Dockerfiles detected from the [plan step](#1-plan) is built using the
+`docker build` command. The workflow automatically changes the working
+directory to where the `*.Dockerfile` file resides.
+The built Docker images are either:
+
+1. Saved as a tarball and uploaded to GitHub
+2. Pushed to the Docker registry
+
+### Charm Resources
+
+The Charm Resources detected from the [plan step](#1-plan) is built by executing
+the `<build-<resourceName>.sh>` script. The workflow automatically changes the
+working directory to where the `<build-<resourceName>.sh>` file resides.
+
+The output charm resource will be uploaded to GitHub.
+
+## Run the integration test
+
+The artifacts built from the [build step](#2-build) are automatically unpacked
+and used in the `tox` integration test.
+
+The tox arguments are then appended accordingly.
+
+### Charm
+
+Tox argument: `--charm-file=./<charm-build-output-file>.charm`
+
+### Rock
+
+The rocks are pushed to the local MicroK8s image registry (localhost:32000)
+using `skopeo copy --insecure-policy --dest-tls-verify=false ...` command.
```

Comment:
Only if the rock is downloaded as an artifact, otherwise it will just use the Github registry.

Reply:
Updated, thank you!